### PR TITLE
[ITG-68] added filters to Task api

### PR DESCRIPTION
**Pull request by Gajkes**
added filters to Task api

### CODE REVIEW

**Review by koszar91 — APPROVED** [review]
Code looks nice. There will be some compatibility isuues with other branches, but probably it won't be your concern XD, you were first to PR.

**Review by gregori0o — APPROVED** [review]

File: bode/bode/resources/tasks/api.py [hunk 1]
```diff
@@ -1,21 +1,44 @@
...
+        if date_from:
+            all_filters.append(Task.due_date >= date_from)
+        if date_to:
+            all_filters.append(Task.due_date <= date_to)
+        if tags:
+            or_filters = []
+            for tag in tags:
+                or_filters.append(Task.tags.any(Tag.name == tag))
+            all_filters.append(or_(*or_filters))
+        return Task.query.filter(*all_filters).all()
```
Comment (john-sonz) [inline bode/bode/resources/tasks/api.py:41]: Bring back the ordering, please!

File: bode/bode/resources/tasks/api.py [hunk 2]
```diff
@@ -1,21 +1,44 @@
...
+        tags = params.get("tags")
+        date_from = params.get("date_from")
+        date_to = params.get("date_to")
+        title = params.get("title")
+
+        all_filters = []
+        if status and status == "todo":
+            all_filters.append(Task.is_done == False)  # noqa
+        if title:
+            all_filters.append(Task.title.like("%" + title + "%"))
```
Comment (john-sonz) [inline bode/bode/resources/tasks/api.py:31]:
```suggestion
            all_filters.append(Task.title.ilike(f"%{title}%"))
```

File: bode/bode/resources/tasks/api.py [hunk 3]
```diff
@@ -1,21 +1,44 @@
...
+        if date_from:
+            all_filters.append(Task.due_date >= date_from)
+        if date_to:
+            all_filters.append(Task.due_date <= date_to)
+        if tags:
+            or_filters = []
+            for tag in tags:
+                or_filters.append(Task.tags.any(Tag.name == tag))
```
Comment (john-sonz) [inline bode/bode/resources/tasks/api.py:39]:
```suggestion
            or_filters = [Task.tags.any(Tag.name == tag) for tag in tags]
```

**Review by john-sonz — CHANGES_REQUESTED** [review]

**Review by john-sonz — APPROVED** [review]
Looks good!

**Pull request closed without merging**